wal/wal.go

```diff
@@ -530,13 +535,15 @@ func (w *WAL) Segments() (m, n int, err error) {
 func (w *WAL) Truncate(i int) error {
 	refs, err := listSegments(w.dir)
 	if err != nil {
+		w.truncateFail.Inc()
 		return err
 	}
 	for _, r := range refs {
 		if r.n >= i {
 			break
 		}
 		if err := os.Remove(filepath.Join(w.dir, r.s)); err != nil {
+			w.truncateFail.Inc()
```

Comment:
not a deal breaker, but to make it more bulletproof against further code changes how about adding this somewhere in the beginning:
```
defer func() {
    if err!=nil{
       w.truncateFail.Inc()
    }
}()
```


this way even if we add more blocks that might return an error wouldn't have to change the logic for the metrics.

Reply:
Thought about that :) I modified now.